### ISSUE TITLE
FlutterFragment uses android.arch.lifecycle.Lifecycle instead of androidx.lifecycle.Lifecycle

### ISSUE THREAD

**Issue by oth-libs** [flutter/flutter#50466]
## Logs
```
flutter doctor
Downloading android-arm-profile/darwin-x64 tools...                 0.8s
Downloading android-arm-release/darwin-x64 tools...                 0.6s
Downloading android-arm64-profile/darwin-x64 tools...               0.6s
Downloading android-arm64-release/darwin-x64 tools...               0.6s
Downloading android-x64-profile/darwin-x64 tools...                 0.7s
Downloading android-x64-release/darwin-x64 tools...                 0.5s
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel dev, v1.15.2, on Mac OS X 10.15.2 19C57, locale en-GB)
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
[✗] Xcode - develop for iOS and macOS
    ✗ Xcode installation is incomplete; a full installation is necessary for iOS development.
      Download at: https://developer.apple.com/xcode/download/
      Or install Xcode via the App Store.
      Once installed, run:
        sudo xcode-select --switch /Applications/Xcode.app/Contents/Developer
        sudo xcodebuild -runFirstLaunch
    ✗ CocoaPods not installed.
        CocoaPods is used to retrieve the iOS and macOS platform side's plugin code that responds to your plugin usage on the Dart side.
        Without CocoaPods, plugins will not work on iOS or macOS.
        For more info, see https://flutter.dev/platform-plugins
      To install:
        sudo gem install cocoapods
[✓] Android Studio (version 3.5)
[✓] Connected device (1 available)

! Doctor found issues in 1 category.
```

I recently switched to the latest version of Flutter from 1.9.x, and since then I found that I need to make a lot of changes because I'm using the facade FlutterView and I just realized that there is no more support for it.

So I'm making the changes and one thing I thought of is switch from FlutterView to FlutterFragment, but it seems that it's using `android.arch.lifecycle.Lifecycle` instead of `androidx.lifecycle.Lifecycle`. 

I have everything setup with androidX and using the latest Flutter version 1.15.
```
pubspec.yaml: 
     module:
          androidX: true
```

```
gradle.properties
android.useAndroidX=true
android.enableJetifier=true
```

What should I do I'm totally blocked by flutter right now.

**Comment by VladyslavBondarenko** [comment]
Hi @oth-libs 
[Here](https://stackoverflow.com/questions/54851489/flutter-projects-android-x-migration-issues) is a good answer but with some outdated versions (for example I use `com.android.tools.build:gradle:3.5.3` in `build.gradle`). Try to follow the steps.
Or as a workaround you can try to [create a new project with androidx](https://flutter.dev/docs/development/androidx-migration#what-about-if-i-cant-use-android-studio) and transfer code from your current one there.
If you still have the error, provide please your `build.gradle` and`app/build.gradle`
Thank you

**Comment by oth-libs** [comment]
@VladyslavBondarenko The project is already androidX, but the classes inside `flutter_embedding_debug-1.0.0` which contains `FlutterFragment` and `FlutterActivity` can't be modified neither by Jetifier nor by Refactor->Migrate to androidX. 
The option to create a new project is unfortunately quite impossible atm. 

```
**build.gradle**
// Generated file. Do not edit.

buildscript {
    repositories {
        google()
        jcenter()
    }

    dependencies {
        classpath 'com.android.tools.build:gradle:3.5.0'
    }
}

allprojects {
    repositories {
        google()
        jcenter()
    }
}

task clean(type: Delete) {
    delete rootProject.buildDir
}
```


```
**app/build.gradle**
def flutterPluginVersion = 'managed'

apply plugin: 'com.android.application'

android {
    compileSdkVersion 28

    compileOptions {
        sourceCompatibility 1.8
        targetCompatibility 1.8
    }

    defaultConfig {
        applicationId "com.example.flu.host"
        minSdkVersion 16
        targetSdkVersion 28
        versionCode 1
        versionName "1.0"
        testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
    }

    buildTypes {
        profile {
            initWith debug
        }
        release {
            // TODO: Add your own signing config for the release build.
            // Signing with the debug keys for now, so `flutter run --release` works.
            signingConfig signingConfigs.debug
        }
    }

}
buildDir = new File(rootProject.projectDir, "../build/host")
dependencies {
    implementation project(':flutter')
    implementation fileTree(dir: 'libs', include: ['*.jar'])
    implementation 'androidx.appcompat:appcompat:1.0.2'
    implementation 'androidx.constraintlayout:constraintlayout:1.1.3'
    androidTestImplementation 'androidx.test:runner:1.1.1'
    androidTestImplementation 'androidx.test.espresso:espresso-core:3.1.1'
    testImplementation 'junit:junit:4.12'
}
```

Thanks!

**Comment by blasten** [comment]
Duplicate of https://github.com/flutter/flutter/issues/39283

**Comment by oth-libs** [comment]
@blasten this doesn't really answer my question. I'm not sure how to proceed can someone please give me a direct way to solve this our app is being blocked by this. thanks!
